spring-integration-core/src/main/java/org/springframework/integration/store/AbstractKeyValueMessageStore.java

```diff
@@ -44,39 +41,61 @@
 
 	protected static final String MESSAGE_GROUP_KEY_PREFIX = "MESSAGE_GROUP_";
 
+	/**
+	 * Represents the time when the message has been added to the store.
+	 * @deprecated since 5.0. This constant isn't used any more.
+	 */
+	@Deprecated
 	protected static final String CREATED_DATE = "CREATED_DATE";
 
 	// MessageStore methods
 
 	@Override
-	public Message<?> getMessage(UUID id) {
-		Message<?> message = getRawMessage(id);
-		if (message != null) {
-			return normalizeMessage(message);
+	public Message<?> getMessage(UUID messageId) {
+		Assert.notNull(messageId, "'messageId' must not be null");
+		Object messageHolder = doRetrieve(MESSAGE_KEY_PREFIX + messageId);
+		if (messageHolder != null) {
+			Assert.isInstanceOf(MessageHolder.class, messageHolder);
```

Comment:
One last though, perhaps we can support `Message<?>` here - it would provide backwards compatibility - messages stored with old versions can still be retrieved.

WDYT?